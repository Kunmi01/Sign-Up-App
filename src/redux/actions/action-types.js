export const ADD_USER_DATA = 'ADD_USER_DATA';
export const ADD_PRIVACY_DATA = 'ADD_PRIVACY_DATA';
