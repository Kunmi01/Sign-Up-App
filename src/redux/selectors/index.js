export const userStepCompleted = ({ userData }) => {
  return Object.keys(userData)
    .filter(key => key !== 'role')
    .every(key => userData[key]);
};

export const privacyStepCompleted = ({ privacyData }) => {
  return Object.keys(privacyData).every(key => privacyData[key] !== null);
};
