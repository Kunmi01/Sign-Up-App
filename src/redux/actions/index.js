import { ADD_USER_DATA, ADD_PRIVACY_DATA } from './action-types';

export const addUserData = payload => {
  return { type: ADD_USER_DATA, payload };
};

export const addPrivacyData = payload => {
  return { type: ADD_PRIVACY_DATA, payload };
};
