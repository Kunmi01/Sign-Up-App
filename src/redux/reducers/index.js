import { ADD_USER_DATA, ADD_PRIVACY_DATA } from '../actions/action-types';

const initialState = {
  userData: {
    name: '',
    role: '',
    email: '',
    password: ''
  },
  privacyData: {
    trayUpdates: null,
    productEmails: null
  }
};

const rootReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_USER_DATA:
      return {
        ...state,
        userData: action.payload
      };
    case ADD_PRIVACY_DATA:
      return {
        ...state,
        privacyData: action.payload
      };
    default:
      return state;
  }
};

export default rootReducer;
