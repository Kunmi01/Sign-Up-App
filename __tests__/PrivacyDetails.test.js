import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ConnectedPrivacyDetails, {
  PrivacyDetails
} from '../src/components/SignUp/components/PrivacyDetails';
import { addPrivacyData } from '../src/redux/actions';

describe('PrivacyDetails', () => {
  const initialState = {
    privacyData: {
      trayUpdates: null,
      productEmails: null
    }
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    store.dispatch = jest.fn();
    container = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedPrivacyDetails />
        </BrowserRouter>
      </Provider>
    );
  });

  it('renders connected(SMART) component', () => {
    expect(container.find(ConnectedPrivacyDetails).exists()).toBe(true);
    expect(toJson(container)).toMatchSnapshot();
  });

  it('prop match with initial store state', () => {
    const privacyDetailsContainer = container.find(PrivacyDetails);
    expect(privacyDetailsContainer.prop('privacyData')).toEqual(
      initialState.privacyData
    );
  });

  it('can submit form', () => {
    const privacyDetailsContainer = container.find(PrivacyDetails);
    renderer.act(() => {
      privacyDetailsContainer.find('input#tray-updates').simulate('change', {
        target: { checked: true }
      });
      privacyDetailsContainer.find('button[type="submit"]').simulate('submit');
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(
      addPrivacyData({
        trayUpdates: true,
        productEmails: false
      })
    );
  });
});
