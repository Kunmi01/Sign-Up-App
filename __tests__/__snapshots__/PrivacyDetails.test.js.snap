// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyDetails renders connected(SMART) component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(PrivacyDetails)>
        <PrivacyDetails
          dispatch={[MockFunction]}
          privacyData={
            Object {
              "productEmails": null,
              "trayUpdates": null,
            }
          }
        >
          <div
            className="privacy-details"
          >
            <div
              className="privacy-details__wrapper"
            >
              <h4
                className="privacy-details__header"
              >
                Privacy Details
              </h4>
              <form
                className="privacy-details__form"
                onSubmit={[Function]}
              >
                <div
                  className="privacy-details__form__input-container privacy-details__form__tray-updates"
                >
                  <label
                    htmlFor="tray-updates"
                  >
                    <input
                      checked={false}
                      id="tray-updates"
                      name="tray-updates"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Receive updates about Tray.io product by email
                  </label>
                </div>
                <div
                  className="privacy-details__form__input-container privacy-details__form__product-emails"
                >
                  <label
                    htmlFor="product-emails"
                  >
                    <input
                      checked={false}
                      id="product-emails"
                      name="product-emails"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    Receive communication by email for other products created by the Tray.io team
                  </label>
                </div>
                <button
                  className="privacy-details__form__submit-button"
                  disabled={false}
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </PrivacyDetails>
      </Connect(PrivacyDetails)>
    </Router>
  </BrowserRouter>
</Provider>
`;
